pkg/daemon/daemon.go

```diff
@@ -759,12 +836,16 @@ func (dn *Daemon) applyNodePtpProfile(runID int, nodeProfile *ptpv1.PtpProfile)
 			lastTransitionResult: event.PTP_NOTSET,
 			handler:              dn.processManager.ptpEventHandler,
 			dn:                   dn,
+			daemon:               dn, // Reference to daemon for hardwareconfig access
 		}
 
 		if pProcess == ptp4lProcessName {
 			if port, ok := (*nodeProfile).PtpSettings["upstreamPort"]; ok && clockType == event.BC {
 				dprocess.tBCAttributes.trIfaceName = port
 			}
+
+			// Prepare cached resources for T-BC processing
+			dprocess.prepareTBCResources()
```

Comment:
Should there be a clock type check? (Might be in the func I haven't checked yet)

Reply:
Done, good catch